### PR TITLE
Add spatial analogs notebook

### LINKED ISSUE #585: Notebook showing how to use spatial analogs
### Description

Create a tutorial showing how to compute spatial analogs with indices computed by xclim.

- [ ] Compute indices from model simulation over historical period
- [ ] Compute indices from model simulation over future period
- [ ] Compute analogs using a couple of metrics
- [ ] Map "distance" according to each metric

We could do this with Climex, to get larger samples for both periods.

### PR DESCRIPTION
**Pull request by aulemahal**
### Pull Request Checklist:
- [x] This PR addresses an already opened issue (for bug fixes / features)
    - This PR fixes #585
- [ ] Tests for the changes have been added (for bug fixes / features)
  - [ ] (If applicable) Documentation has been added / updated (for bug fixes / features)
- [x] HISTORY.rst has been updated (with summary of main changes)
  - [x] Link to issue (:issue:`number`) and pull request (:pull:`number`) has been added
- [ ] `bumpversion patch` has been called on this branch
- [x] The relevant author information has been added to `.zenodo.json`

### What kind of change does this PR introduce?

* Adds a notebook showcasing simple spatial analogs. This uses a new test dataset, so we need to merge that branch in `xclim-testdata` before this one.

With only one target and one reference, there's not much you can do more than showing different indices and metrics...

### Does this PR introduce a breaking change?
No.

### Other information:
* Minor change to the doc conf, the path to the scipy intersphinx changed. Sphinx detected it automatically, but changing it is cleaner.

### CODE REVIEW

**Comment by huard** [comment]
@aulemahal Révisé dans reviewNB

File: HISTORY.rst [hunk 1]
```diff
@@ -16,11 +16,13 @@ New features and enhancements
 * All indicators now have indexing enabled, except those computing statistics on spells. (:issue:`1069`, :pull:`1070`).
 * A convenience function for returning the version numbers for relevant xclim dependencies (``xclim.testing.show_versions``) is now offered. (:pull:`1073`).
   * A CLI version of this function is also available from the command line: `$ xclim show_version_info`. (:pull:`1073`).
+* Added a notebook showcasing some simple examples of Spatial Analogues. (:issue:`585`, :pull:`1075`).
 
 Internal changes
 ^^^^^^^^^^^^^^^^
 * Typing syntax has been updated within pre-commit via `isort`. Pre-commit hooks now append `from __future__ import annotations` to all python module imports for backwards compatibility. (:issue:`1065`, :pull:`1071`)
 * `isort` project configurations are now set in `setup.cfg`. (:pull:`1071`).
+>>>>>>> master
```
Comment (huard) [inline HISTORY.rst:25]:
```suggestion

```

**Review by huard — APPROVED** [review]
Voir commentaires dans ReviewNB.

**Comment by aulemahal** [comment]
@Zeitsperre David and I found it a bit weird to add `from __future__ import annotations` in notebooks when not needed. But I haven't found a good way to skip that step in the `nbqa-isort` hook. However, it seemed to me that the added value of running isort on notebooks can't be that high? The last commit removes it, but what's your opinion on the matter?

**Comment by Zeitsperre** [comment]
> @Zeitsperre David and I found it a bit weird to add `from __future__ import annotations` in notebooks when not needed. But I haven't found a good way to skip that step in the `nbqa-isort` hook. However, it seemed to me that the added value of running isort on notebooks can't be that high? The last commit removes it, but what's your opinion on the matter?

I think there's a way to disable the added import for notebooks. I feel like it's nice to have (we want to showcase clean code examples), not mission-critical, but why not profit from it?

**Comment by Zeitsperre** [comment]
I think https://github.com/Ouranosinc/xclim/pull/1075/commits/e54ae4dc803c2e0ea1943241ced554f68a83288a is a reasonable solution to the problem. If not, I'm fine with removing nbqa-isort.

**Comment by MarcoGorelli** [comment]
> I think there's a way to disable the added import for notebooks.

Hey - if you want this in a configuration file, you could add a `pyproject.toml` file with
```
[tool.nbqa.addopts]
isort = [
    "--add-import="
]
```
 - this should override the `--add-import` option passed to plain `isort` when running it on notebooks

**Comment by Zeitsperre** [comment]
@MarcoGorelli Thanks for the suggestion! I've noticed a big push from a lot of Python CQA library developers to migrate to `pyproject.toml`. Perhaps we'll look into doing that here in the near future.

Oh, also thanks for developing Pandas (and NBQA)! None of this would possible without it!

**Comment by MarcoGorelli** [comment]
😄 happy to help (though for pandas I can claim credit only for an invisibly tiny minute fraction of the product)

**Pull request merged**